### PR TITLE
[CI] Add backport action for automated backports

### PR DESCRIPTION
**Pull request by kavilla**
### Description
Adding VachaShah/backport@v1.1.4 for automating backport PRs.

Signed-off-by: Kawika Avilla <kavilla414@gmail.com>
 
### Issues Resolved
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1211
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
    - [x] `yarn test:jest`
    - [x] `yarn test:jest_integration`
    - [x] `yarn test:ftr`
~~- [ ] New functionality has been documented.~~
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v1.5.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
+
+      - name: Backport
+        uses: VachaShah/backport@v1.1.4
```
[the remark this answers is not in the document]
Comment (VachaShah) [reply .github/workflows/backport.yml:25]: This fork has a custom branch feature where we can give backport branch names to better manage branch permissions. I tried PRing this into the original but had to release it on the fork due to lack of activity.

File: .github/workflows/backport.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      pull-requests: write
+    name: Backport
+    steps:
+      - name: GitHub App token
+        id: github_app_token
+        uses: tibdex/github-app-token@v1.5.0
+        with:
+          app_id: ${{ secrets.APP_ID }}
+          private_key: ${{ secrets.APP_PRIVATE_KEY }}
+          installation_id: 22958780
```
Comment (tmarkley) [inline .github/workflows/backport.yml:22]: Where does the id come from?

File: .github/workflows/backport.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (VachaShah) [reply .github/workflows/backport.yml:25]: Sure, I am also going to release the fork I have as a GHA and will keep it up-to-date with the original GHA.

File: .github/workflows/backport.yml (hunk 2 above)
Comment (VachaShah) [reply .github/workflows/backport.yml:22]: Its the installation ID for the `opensearch-trigger-bot` app.

**Comment by tmarkley** [comment]
Can you update the reference to the backport process in the [CONTRIBUTING](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/CONTRIBUTING.md) file?

**Comment by kavilla** [comment]
> Can you update the reference to the backport process in the [CONTRIBUTING](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/CONTRIBUTING.md) file?

Sure! What would you be your recommendation? Right now I think non-maintainers can't add labels so should be just specific in that a maintainer will add a label to the PR?

File: .github/workflows/backport.yml (hunk 2 above)
Comment (tmarkley) [reply .github/workflows/backport.yml:22]: Rocky can we add a comment for that here?

File: .github/workflows/backport.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (tmarkley) [reply .github/workflows/backport.yml:25]: Can we add a comment with a link to the repo?

**Comment by tmarkley** [comment]
> > Can you update the reference to the backport process in the [CONTRIBUTING](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/CONTRIBUTING.md) file?
> 
> Sure! What would you be your recommendation? Right now I think non-maintainers can't add labels so should be just specific in that a maintainer will add a label to the PR?

Yeah I would say just add instructions for the maintainers so that it's clear what the process is.

File: .github/workflows/backport.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (kavilla) [reply .github/workflows/backport.yml:28]: Done!

File: .github/workflows/backport.yml (hunk 2 above)
Comment (kavilla) [reply .github/workflows/backport.yml:23]: Done!

**Review by tmarkley — APPROVED** [review]
Looks great, thanks!

**Review by ananzh — APPROVED** [review]

**Pull request merged**